Application/Education/Certificate/Generator/Service/DataCertificate/IDataESBD.php

```diff
@@ -0,0 +1,969 @@
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setEsbdMsHjFsLernen(Data $Data, TblConsumer $tblConsumerCertificate)
+    {
+
+        $tblCertificate = $Data->createCertificate('Oberschule Halbjahreszeugnis', 'Förderschwerpunkt Lernen', 'ESBD\EsbdMsHjFsLernen', $tblConsumerCertificate);
+        if ($tblCertificate) {
+            if ($Data->getTblSchoolTypeSecondary()) {
+                $Data->updateCertificate($tblCertificate, $Data->getTblCertificateTypeHalfYear(), $Data->getTblSchoolTypeSecondary(), null, false, true);
+                if (!$Data->getCertificateLevelAllByCertificate($tblCertificate)) {
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypeSecondary(), '10'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                }
+            }
+            // Begrenzung des Bemerkungsfeld
+            $FieldName = 'Remark';
+            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+                $Data->createCertificateField($tblCertificate, $FieldName, 700);
+            }
+
+            // Seite 2
+            // Begrenzung der Dialoge
+            self::setESBDSecondPageLength($Data, $tblCertificate);
+        }
+        if ($tblCertificate && !$Data->getCertificateGradeAll($tblCertificate)) {
+            $Data->setCertificateGrade($tblCertificate, 'PN', 1, 1);
+        }
+        if ($tblCertificate && !$Data->getCertificateSubjectAll($tblCertificate)) {
+            $Data->setCertificateSubject($tblCertificate, 'DE', 1, 1);
+            $Data->setCertificateSubject($tblCertificate, 'EN', 1, 2);
+            $Data->setCertificateSubject($tblCertificate, 'KU', 1, 3);
+            $Data->setCertificateSubject($tblCertificate, 'MU', 1, 4);
+            $Data->setCertificateSubject($tblCertificate, 'GE', 1, 5);
+            $Data->setCertificateSubject($tblCertificate, 'GK', 1, 6);
+            $Data->setCertificateSubject($tblCertificate, 'GEO', 1, 7);
+            $Data->setCertificateSubject($tblCertificate, 'WTH', 1, 8);
+
+            $Data->setCertificateSubject($tblCertificate, 'MA', 2, 1);
+            $Data->setCertificateSubject($tblCertificate, 'BIO', 2, 2);
+            $Data->setCertificateSubject($tblCertificate, 'CH', 2, 3);
+            $Data->setCertificateSubject($tblCertificate, 'PH', 2, 4);
+            $Data->setCertificateSubject($tblCertificate, 'SPO', 2, 5);
+            $Data->setCertificateSubject($tblCertificate, 'RE/e', 2, 6);
+            $Data->setCertificateSubject($tblCertificate, 'TC', 2, 7);
+            $Data->setCertificateSubject($tblCertificate, 'INF', 2, 8);
+        }
+    }
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setEsbdMsJFsLernen(Data $Data, TblConsumer $tblConsumerCertificate)
```

Comment:
Codacy found an issue: [The method setEsbdMsJFsLernen() has an NPath complexity of 1233. The configured NPath complexity threshold is 200.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)